indexing-service/src/main/java/io/druid/indexing/worker/WorkerTaskMonitor.java

```diff
@@ -200,7 +227,8 @@ public void stop()
   {
     try {
       pathChildrenCache.close();
-      exec.shutdown();
+      exec.shutdownNow();
```

Comment:
I have to interrupt the thread, because it's likely going to be blocking waiting for the task future to complete. It *should* be okay to interrupt even if it's doing something else, since this stuff is meant to work even if it dies at any time.